src/os/bluestore/BlueStore.h

```diff
@@ -1086,276 +1086,281 @@ class BlueStore : public ObjectStore,
+        &Buffer::lru_item> > list_t;
+    list_t lru;
 
-    buffer_lru_list_t buffer_lru;
-    uint64_t buffer_size = 0;
+    LruBufferCacheShard(CephContext *cct) : BufferCacheShard(cct) {}
 
-  public:
-    LRUCache(CephContext* cct) : Cache(cct) {}
-    uint64_t _get_num_onodes() override {
-      return onode_lru.size();
-    }
-    void _add_onode(OnodeRef& o, int level) override {
-      if (level > 0)
-	onode_lru.push_front(*o);
-      else
-	onode_lru.push_back(*o);
-    }
-    void _rm_onode(OnodeRef& o) override {
-      auto q = onode_lru.iterator_to(*o);
-      onode_lru.erase(q);
-    }
-    void _touch_onode(OnodeRef& o) override;
-
-    uint64_t _get_buffer_bytes() override {
-      return buffer_size;
-    }
-    void _add_buffer(Buffer *b, int level, Buffer *near) override {
+    void _add(Buffer *b, int level, Buffer *near) override {
       if (near) {
-	auto q = buffer_lru.iterator_to(*near);
-	buffer_lru.insert(q, *b);
+        auto q = lru.iterator_to(*near);
+        lru.insert(q, *b);
       } else if (level > 0) {
-	buffer_lru.push_front(*b);
+        lru.push_front(*b);
       } else {
-	buffer_lru.push_back(*b);
+        lru.push_back(*b);
       }
-      buffer_size += b->length;
+      buffer_bytes += b->length;
     }
-    void _rm_buffer(Buffer *b) override {
-      ceph_assert(buffer_size >= b->length);
-      buffer_size -= b->length;
-      auto q = buffer_lru.iterator_to(*b);
-      buffer_lru.erase(q);
+    void _rm(Buffer *b) override {
+      ceph_assert(buffer_bytes >= b->length);
+      buffer_bytes -= b->length;
+      auto q = lru.iterator_to(*b);
+      lru.erase(q);
     }
-    void _move_buffer(Cache *src, Buffer *b) override {
-      src->_rm_buffer(b);
-      _add_buffer(b, 0, nullptr);
+    void _move(BufferCacheShard *src, Buffer *b) override {
+      src->_rm(b);
+      _add(b, 0, nullptr);
     }
-    void _adjust_buffer_size(Buffer *b, int64_t delta) override {
-      ceph_assert((int64_t)buffer_size + delta >= 0);
-      buffer_size += delta;
+    void _adjust_size(Buffer *b, int64_t delta) override {
+      ceph_assert((int64_t)buffer_bytes + delta >= 0);
+      buffer_bytes += delta;
     }
-    void _touch_buffer(Buffer *b) override {
-      auto p = buffer_lru.iterator_to(*b);
-      buffer_lru.erase(p);
-      buffer_lru.push_front(*b);
+    void _touch(Buffer *b) override {
+      auto p = lru.iterator_to(*b);
+      lru.erase(p);
+      lru.push_front(*b);
       _audit("_touch_buffer end");
     }
 
-    void _trim_onodes_to(uint64_t max) override;
-    void _trim_buffers_to(uint64_t max) override;
+    void _trim_to(uint64_t max) override;
 
-    void add_stats(uint64_t *onodes, uint64_t *extents,
-		   uint64_t *blobs,
-		   uint64_t *buffers,
-		   uint64_t *bytes) override {
+    void add_stats(uint64_t *extents,
+                   uint64_t *blobs,
+                   uint64_t *buffers,
+                   uint64_t *bytes) override {
       std::lock_guard l(lock);
```

Comment:
similarly we can get rid off the lock if tracking lru size externally using atomic

Reply:
Agreed. Done!